eng/packages/http-client-csharp-mgmt/emitter/src/resource-detection.ts

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import {
+  CodeModel,
+  InputClient,
+  InputModelType
+} from "@typespec/http-client-csharp";
+import {
+  calculateResourceTypeFromPath,
+  ResourceMetadata
+} from "./resource-metadata.js";
+import { DecoratorInfo } from "@azure-tools/typespec-client-generator-core";
+
+// https://github.com/microsoft/typespec/blob/main/packages/rest/README.md#parentresource
```

Comment:
consider to put the const string to a central place

Reply:
I will move them into the `sdk-context-options.ts` file